src/lynx-data-parser.ts

```diff
@@ -66,7 +66,12 @@ function parseResults(message: string): LynxResults {
 export function parseLynxPacket(
   message: string
 ): { isDirective: boolean; data: LynxResults | LynxDirective } {
-  const isDirective = message.startsWith("*");
+  // Check if directive 
+  let isDirective = false;
```

Comment:
There is no reason to make this a let and set it twice.

Leave `const isDirective = message.startsWith("*");`

And then do

`message = isDirective ? message.substr(1) : message;`

and then this 5 lines is down to 2

alternately (actually would be better architecture) you could just trim off the * in the parseDirective method and avoid any if forks in this function

Reply:
Yeah do that.

`mesage = message.substr(1);`

in the `parseDirective` function